### PR TITLE
Updating UnsupportedPythonError(f"sunpy does not support Python < {__…

### PR DESCRIPTION
**Pull request by dodoextinct**
…minimum_python_version__}")

There is a syntax error when the exception is being raised.

### Description

Fixes #<Issue Number>

### CODE REVIEW

**Comment by nabobalis** [comment]
Will that not still error?

**Comment by dodoextinct** [comment]
> Will that not still error?

The syntax error is removed I think.

**Comment by nabobalis** [comment]
Have you checked?

**Comment by dodoextinct** [comment]
> Have you checked?

Yes, I ran it locally, it was removed.

**Comment by nabobalis** [comment]
> > Have you checked?
> 
> Yes, I ran it locally, it was removed.

With what Python version and how did the error print?

**Comment by dodoextinct** [comment]
> > > Have you checked?
> > 
> > 
> > Yes, I ran it locally, it was removed.
> 
> With what Python version and how did the error print?

Error:
`File "./__init__.py", line 25
    raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}")
                                                                                               ^
SyntaxError: invalid syntax
`

**Comment by nabobalis** [comment]
> > > > Have you checked?
> > > 
> > > 
> > > Yes, I ran it locally, it was removed.
> > 
> > 
> > With what Python version and how did the error print?
> 
> Error:
> `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `

Sorry, I meant how does it error now?

**Comment by dodoextinct** [comment]
> > > > > Have you checked?
> > > > 
> > > > 
> > > > Yes, I ran it locally, it was removed.
> > > 
> > > 
> > > With what Python version and how did the error print?
> > 
> > 
> > Error:
> > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> 
> Sorry, I meant how does it error now?

`Traceback (most recent call last):
  File "./__init__.py", line 25, in <module>
    raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}")
__main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__}
`

I think it can be improved more.

**Comment by nabobalis** [comment]
> > > > > > Have you checked?
> > > > > 
> > > > > 
> > > > > Yes, I ran it locally, it was removed.
> > > > 
> > > > 
> > > > With what Python version and how did the error print?
> > > 
> > > 
> > > Error:
> > > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> > 
> > 
> > Sorry, I meant how does it error now?
> 
> `Traceback (most recent call last): File "./__init__.py", line 25, in <module> raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}") __main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__} `
> 
> I think it can be improved more.

Yes, the `f` was doing string formatting on the variable `__minimum_python_version__`, so you would also need to change that to `.format`. 

What python version are you using? 3.5?

**Comment by dodoextinct** [comment]
> > > > > > > Have you checked?
> > > > > > 
> > > > > > 
> > > > > > Yes, I ran it locally, it was removed.
> > > > > 
> > > > > 
> > > > > With what Python version and how did the error print?
> > > > 
> > > > 
> > > > Error:
> > > > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> > > 
> > > 
> > > Sorry, I meant how does it error now?
> > 
> > 
> > `Traceback (most recent call last): File "./__init__.py", line 25, in <module> raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}") __main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__} `
> > I think it can be improved more.
> 
> Yes, the `f` was doing string formatting on the variable `__minimum_python_version__`, so you would also need to change that to `.format`.
> 
> What python version are you using? 3.5?

Yes I used `.format`, I didn't commit though. 
Also, I updated conda python version to 3.7 but somehow it's still showing this error.

**Comment by nabobalis** [comment]
> > > > > > > > Have you checked?
> > > > > > > 
> > > > > > > 
> > > > > > > Yes, I ran it locally, it was removed.
> > > > > > 
> > > > > > 
> > > > > > With what Python version and how did the error print?
> > > > > 
> > > > > 
> > > > > Error:
> > > > > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> > > > 
> > > > 
> > > > Sorry, I meant how does it error now?
> > > 
> > > 
> > > `Traceback (most recent call last): File "./__init__.py", line 25, in <module> raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}") __main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__} `
> > > I think it can be improved more.
> > 
> > 
> > Yes, the `f` was doing string formatting on the variable `__minimum_python_version__`, so you would also need to change that to `.format`.
> > What python version are you using? 3.5?
> 
> Yes I used `.format`, I didn't commit though.
> Also, I updated conda python version to 3.7 but somehow it's still showing this error.

That makes no sense since 3.7 is supported. I wonder if its the __pycache__ files in your folder causing that.

**Comment by dodoextinct** [comment]
> > > > > > > > > Have you checked?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Yes, I ran it locally, it was removed.
> > > > > > > 
> > > > > > > 
> > > > > > > With what Python version and how did the error print?
> > > > > > 
> > > > > > 
> > > > > > Error:
> > > > > > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> > > > > 
> > > > > 
> > > > > Sorry, I meant how does it error now?
> > > > 
> > > > 
> > > > `Traceback (most recent call last): File "./__init__.py", line 25, in <module> raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}") __main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__} `
> > > > I think it can be improved more.
> > > 
> > > 
> > > Yes, the `f` was doing string formatting on the variable `__minimum_python_version__`, so you would also need to change that to `.format`.
> > > What python version are you using? 3.5?
> > 
> > 
> > Yes I used `.format`, I didn't commit though.
> > Also, I updated conda python version to 3.7 but somehow it's still showing this error.
> 
> That makes no sense since 3.7 is supported. I wonder if its the **pycache** files in your folder causing that.

Yeah, actually that's why I am confused. What should I do?

**Comment by nabobalis** [comment]
> > > > > > > > > > Have you checked?
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > Yes, I ran it locally, it was removed.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > With what Python version and how did the error print?
> > > > > > > 
> > > > > > > 
> > > > > > > Error:
> > > > > > > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> > > > > > 
> > > > > > 
> > > > > > Sorry, I meant how does it error now?
> > > > > 
> > > > > 
> > > > > `Traceback (most recent call last): File "./__init__.py", line 25, in <module> raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}") __main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__} `
> > > > > I think it can be improved more.
> > > > 
> > > > 
> > > > Yes, the `f` was doing string formatting on the variable `__minimum_python_version__`, so you would also need to change that to `.format`.
> > > > What python version are you using? 3.5?
> > > 
> > > 
> > > Yes I used `.format`, I didn't commit though.
> > > Also, I updated conda python version to 3.7 but somehow it's still showing this error.
> > 
> > 
> > That makes no sense since 3.7 is supported. I wonder if its the **pycache** files in your folder causing that.
> 
> Yeah, actually that's why I am confused. What should I do?

You can clean out the repo using `git clean -xfd`. WARNING THIS WILL REMOVE ANY AND ALL UNTRACKED FILES.

**Comment by dodoextinct** [comment]
> > > > > > > > > > > Have you checked?
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > Yes, I ran it locally, it was removed.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > With what Python version and how did the error print?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Error:
> > > > > > > > `File "./__init__.py", line 25 raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}") ^ SyntaxError: invalid syntax `
> > > > > > > 
> > > > > > > 
> > > > > > > Sorry, I meant how does it error now?
> > > > > > 
> > > > > > 
> > > > > > `Traceback (most recent call last): File "./__init__.py", line 25, in <module> raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}") __main__.UnsupportedPythonError: sunpy does not support Python < {__minimum_python_version__} `
> > > > > > I think it can be improved more.
> > > > > 
> > > > > 
> > > > > Yes, the `f` was doing string formatting on the variable `__minimum_python_version__`, so you would also need to change that to `.format`.
> > > > > What python version are you using? 3.5?
> > > > 
> > > > 
> > > > Yes I used `.format`, I didn't commit though.
> > > > Also, I updated conda python version to 3.7 but somehow it's still showing this error.
> > > 
> > > 
> > > That makes no sense since 3.7 is supported. I wonder if its the **pycache** files in your folder causing that.
> > 
> > 
> > Yeah, actually that's why I am confused. What should I do?
> 
> You can clean out the repo using `git clean -xfd`. WARNING THIS WILL REMOVE ANY AND ALL UNTRACKED FILES.

It's still showing the same error.

**Comment by nabobalis** [comment]
So you are using python 3.7 from conda and it errors about the wrong python version?

**Comment by dodoextinct** [comment]
> So you are using python 3.7 from conda and it errors about the wrong python versi

I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.

**Comment by nabobalis** [comment]
> > So you are using python 3.7 from conda and it errors about the wrong python versi
> 
> I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.

When you start python, does it say 3.7?

**Comment by dodoextinct** [comment]
> > > So you are using python 3.7 from conda and it errors about the wrong python versi
> > 
> > 
> > I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.
> 
> When you start python, does it say 3.7?

It says python 3.7 when I saw it in `conda list`.

**Comment by nabobalis** [comment]
> > > > So you are using python 3.7 from conda and it errors about the wrong python versi
> > > 
> > > 
> > > I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.
> > 
> > 
> > When you start python, does it say 3.7?
> 
> It says python 3.7 when I saw it in `conda list`.

But if you run python, the version is 3.7?

File: sunpy/__init__.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class UnsupportedPythonError(Exception):
 
 
 if sys.version_info < tuple((int(val) for val in __minimum_python_version__.split('.'))):
-    raise UnsupportedPythonError(f"sunpy does not support Python < {__minimum_python_version__}")
+    raise UnsupportedPythonError("sunpy does not support Python < {__minimum_python_version__}")
```
Comment (nabobalis) [inline sunpy/__init__.py]:
```suggestion
    raise UnsupportedPythonError("sunpy does not support Python < {}".format(__minimum_python_version__)
```

**Comment by dodoextinct** [comment]
> > > > > So you are using python 3.7 from conda and it errors about the wrong python versi
> > > > 
> > > > 
> > > > I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.
> > > 
> > > 
> > > When you start python, does it say 3.7?
> > 
> > 
> > It says python 3.7 when I saw it in `conda list`.
> 
> But if you run python, the version is 3.7?

No, it's python 3.5.

**Comment by nabobalis** [comment]
> > > > > > So you are using python 3.7 from conda and it errors about the wrong python versi
> > > > > 
> > > > > 
> > > > > I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.
> > > > 
> > > > 
> > > > When you start python, does it say 3.7?
> > > 
> > > 
> > > It says python 3.7 when I saw it in `conda list`.
> > 
> > 
> > But if you run python, the version is 3.7?
> 
> No, it's python 3.5.

Then you didn't update to 3.7?

**Comment by dodoextinct** [comment]
> > > > > > > So you are using python 3.7 from conda and it errors about the wrong python versi
> > > > > > 
> > > > > > 
> > > > > > I had python 3.5 installed earlier. So before setting up sunpy, I installed miniconda and python 3.7 along with it and then I installed sunpy according to the installation guide.
> > > > > 
> > > > > 
> > > > > When you start python, does it say 3.7?
> > > > 
> > > > 
> > > > It says python 3.7 when I saw it in `conda list`.
> > > 
> > > 
> > > But if you run python, the version is 3.7?
> > 
> > 
> > No, it's python 3.5.
> 
> Then you didn't update to 3.7?

I installed conda with python 3.7, won't that do? Do I have to update system's default python version also?

**Comment by nabobalis** [comment]
> I installed conda with python 3.7, won't that do? Do I have to update system's default python version also?

Well you need to create a virtual environment and activate it to use the python version included with conda. Here are some instructions: https://towardsdatascience.com/getting-started-with-python-environments-using-conda-32e9f2779307

**Comment by dodoextinct** [comment]
> > I installed conda with python 3.7, won't that do? Do I have to update system's default python version also?
> 
> Well you need to create a virtual environment and activate it to use the python version included with conda. Here are some instructions: https://towardsdatascience.com/getting-started-with-python-environments-using-conda-32e9f2779307

I have already done it, but I can try once more. Thank you.

**Comment by dodoextinct** [comment]
> > > I installed conda with python 3.7, won't that do? Do I have to update system's default python version also?
> > 
> > 
> > Well you need to create a virtual environment and activate it to use the python version included with conda. Here are some instructions: https://towardsdatascience.com/getting-started-with-python-environments-using-conda-32e9f2779307
> 
> I have already done it, but I can try once more. Thank you.

@nabobalis about this pull request, since the error will be shown in versions below 3.6, the error formatting should also use `.format` than using `f`, then only it will show?

**Comment by nabobalis** [comment]
Should do.

**Comment by nabobalis** [comment]
Thanks @dodoextinct

**Pull request merged**